src/cbica_toolkit/src/cbicaUtilities.cpp

```diff
@@ -751,13 +749,16 @@ namespace cbica
     }
 #else
     //! Initialize pointers to file and user names
-    char path[PATH_MAX];
-    if (::readlink("/proc/self/exe", path, sizeof(path) - 1) == -1)
-      //path = dirname(path);
-      std::cerr << "[getFullPath()] Error during getting full path..";
+     char path[PATH_MAX];
+     ssize_t count = readlink("/proc/self/exe", path, PATH_MAX);
+     if (count == -1)
+       std::cerr << "[getFullPath()] Error during getting full path..";
+    std::string appPath = std::string( path, (count > 0) ? count : 0 );
+    path[0] = '\0';
+    return appPath;
 #endif
 
-    std::string return_string = std::string(path);
+    return_string = std::string(path);
     path[0] = '\0';
```

Comment:
This line still doesn't do anything? I'm just mentioning it in case there is some logic error. Everything you do to ```path``` after ```return_string = std::string(path);``` doesn't affect ```return_string```.

Reply:
I am not doing any changes to ```return_string```; I am merely setting the pointer ```path``` to null.

For your reference, give [this](http://www.cs.ecu.edu/karl/2530/spr17/Notes/C/String/nullterm.html) a read.